engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1738,38 +1831,78 @@ public Class<?> visit(MethodCallExpr n, VisitArgs printer) {
         return calculateMethodReturnTypeUsingGenerics(scope, method, expressionTypes, parameterizedTypes);
     }
 
-    private void checkPyNumbaVectorizedFunc(MethodCallExpr n, Expression[] expressions, Class<?>[] expressionTypes) {
-        // numba vectorized functions return arrays of primitive types. This will break the generated expression
-        // evaluation code that expects singular values. This check makes sure that numba vectorized functions must be
-        // used alone (or with cast only) as the entire expression.
+    private void prepareVectorization(MethodCallExpr n, Expression[] expressions, Class<?>[] expressionTypes) {
+        // find the Python CallableWrapper
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        QueryScopeParam<?>[] params = queryScope.getParams(queryScope.getParamNames());
+
+        Optional<QueryScopeParam<?>> paramOptional =
+                Arrays.stream(params)
+                        .filter(p -> p.getValue() instanceof CallableWrapper && p.getName().equals(n.getNameAsString()))
+                        .findFirst();
+
+        if (paramOptional.isPresent()) {
+            QueryScopeParam<?> param = paramOptional.get();
+            CallableWrapper callableWrapper = (CallableWrapper) param.getValue();
+            try {
+                checkVectorizability(n, expressions, expressionTypes, callableWrapper);
+                callableWrapper.setVectorizable(true);
+                if (!callableWrapper.isVectorized()) {
+                    log.info().append("Python function call ").append(n.toString()).append(" is auto-vectorizable")
```

Comment:
I assume these log messages should move to `debug` level? Applies to L1860, as well.